psydac/fem/splines.py

```diff
@@ -469,6 +469,13 @@ def compute_histopolant( self, values, field ):
         c[0:n] = self._histopolator.solve( values )
         c.update_ghost_regions()
 
+    def refine(self, ncells):
+        if ncells == len(self.breaks)-1:
+            return self
+        grid = np.linspace(self.breaks[0], self.breaks[-1], ncells+1)
+        return SplineSpace(self.degree, grid=grid, multiplicity=self.multiplicity, parent_multiplicity=self.parent_multiplicity,
+                            periodic=self.periodic, dirichlet=self.dirichlet, basis=self.basis, pads=self.pads)
```

Comment:
yes, it needs to be updated, here we assume that we use uniform splines

Reply:
@saidctb Even if we use uniform splines, this can only work if `ncells` is a multiple of `self.ncells`. If this is not the case, `self` will not be a subspace of the refined space just created, and this construction will be meaningless.

@campospinto @FrederikSchnack What do you suggest here?

Reply:
yes, this needs to be improved to make sure that the space is a subspace of the refined one

Reply:
Fixed in 64e6da6